status/versions.go

```diff
@@ -108,6 +108,121 @@ func validateVersion(requiredVersion string, installedVersion string) bool {
 	return false
 }
 
+// checkMeshVersion check mesh if its version is compatible with kiali
+func CheckMeshVersion(meshName string, meshVersion string, kialiVersion string) bool {
+	ok := false
+	if meshName == istioProductNameUnknown {
+		return ok
+	}
+	if strings.Contains(meshName, "Istio") {
+		ok = checkIstioVersion(meshVersion, kialiVersion)
+		return ok
+	} else if strings.Contains(meshName, "Maistra") {
+		ok = checkMaistraVersion(meshVersion, kialiVersion)
+		return ok
+	} else if strings.Contains(meshName, "OpenShift Service Mesh") {
+		ok = checkOSSMVersion(meshVersion, kialiVersion)
+		return ok
+	}
+	return ok
+}
+
+// checkOSSMVersion check OpenShift Service Mesh if its version is compatible with kiali
+func checkOSSMVersion(ossmVersion string, kialiVersion string) bool {
+	ok := false
+	Versions, err := config.NewVersions()
+
+	if err != nil {
+		log.Warningf("Can not load version file.%v", err)
+	}
+
+	for _, version := range Versions {
+		if version.MeshName == "OpenShift Service Mesh" {
+			for _, versions := range version.VersionRange {
+				if ossmVersion == strings.TrimSpace(versions.MeshVersion) && kialiVersion == strings.TrimSpace(versions.KialiMinimumVersion) {
+					ok = true
+					break
+				}
+			}
+		}
+	}
+
+	return ok
+}
+
+// checkMaistraVersion check Maistra if its version is compatible with kiali
+func checkMaistraVersion(maistraVersion string, kialiVersion string) bool {
+	ok := false
+	Versions, err := config.NewVersions()
+
+	if err != nil {
+		log.Warningf("Can not load version file.%v", err)
+	}
+
+	for _, version := range Versions {
+		if version.MeshName == "Maistra" {
+			for _, versions := range version.VersionRange {
+				if maistraVersion == strings.TrimSpace(versions.MeshVersion) && kialiVersion == strings.TrimSpace(versions.KialiMinimumVersion) {
+					ok = true
+					break
+				}
+			}
+		}
+	}
+
+	return ok
+}
+
+// checkIstioVersion check istio if its version is compatible with kiali
+func checkIstioVersion(istioVersion string, kialiVersion string) bool {
+	ok := false
```

Comment:
I haven't looked at what happens when this function returns false, but if we disrupt the user session at all, I think that would be a mistake. I think regardless of whether we determine the current version of Istio is "supported" or not, the user should be allowed to continue as-is normally. We should not log them out, we should not restrict access or functionality at all - they should be able to use Kiali normally as before. This new stuff should only result in a visible "warning" to the user. That warning should say something to the effect of, "Hey, if you see something funny happen, the first thing you should do is upgrade Kiali or Istio because the error is probably due to the fact you are running an obsolete/deprecated/unsupported Kiali compared to the Istio impementation you have installed"

In other words, it shouldn't matter what this function returns - the user should continue normally.

In that case (assuming the above is true - the user session is not affected no matter what this check returns), then I'm of no opinion what this should return if we cannot determine whether kiali supports the installed istio or not.

Reply:
Actually I followed this rule which this task will not affect kiali normally running status, which means no matter the default ok is true or false, the user should continue normally.